dropwizard-validation/src/main/java/io/dropwizard/validation/ConstraintViolations.java

```diff
@@ -15,17 +15,21 @@ private ConstraintViolations() { /* singleton */ }
 
     public static <T> String format(ConstraintViolation<T> v) {
         if (v.getConstraintDescriptor().getAnnotation() instanceof ValidationMethod) {
-            final ImmutableList<Path.Node> nodes = ImmutableList.copyOf(v.getPropertyPath());
-            final ImmutableList<Path.Node> usefulNodes = nodes.subList(0, nodes.size() - 1);
-            final String msg = v.getMessage().startsWith(".") ? "%s%s" : "%s %s";
-            return String.format(msg,
-                                 Joiner.on('.').join(usefulNodes),
-                                 v.getMessage()).trim();
+            return validationMethodFormatted(v);
         } else {
             return String.format("%s %s", v.getPropertyPath(), v.getMessage());
         }
     }
 
+    public static <T> String validationMethodFormatted(ConstraintViolation<T> v) {
```

Comment:
We probably can reuse `Joiner` and don't use `String.format`.
````java
public static <T> String validationMethodFormatted(ConstraintViolation<T> v) {
        final ImmutableList<Path.Node> nodes = ImmutableList.copyOf(v.getPropertyPath());
        String usefulNodes = JOINER.join(nodes.subList(0, nodes.size() - 1));
        String msg = usefulNodes + (v.getMessage().startsWith(".") ? "" : " ") + v.getMessage();
        return msg.trim();
    }
````
Kindly provided by you microbenchmark gives this on my machine:
````
Benchmark                                    Mode  Cnt     Score      Error  Units
ConstraintViolationBenchmark.objViolation    avgt    5   711.064 ±   22.911  ns/op
ConstraintViolationBenchmark.paramViolation  avgt    5  7400.960 ± 2268.245  ns/op

Benchmark                                    Mode  Cnt     Score     Error  Units
ConstraintViolationBenchmark.objViolation    avgt    5   693.016 ±  10.078  ns/op
ConstraintViolationBenchmark.paramViolation  avgt    5  6585.522 ± 815.698  ns/op
````